pkg/search/postgres/store.go

```diff
@@ -256,3 +256,31 @@ func (s *GenericStore[T, PT]) GetByQuery(ctx context.Context, query *v1.Query) (
 	}
 	return rows, nil
 }
+
+// GetIDs returns all the IDs for the store.
+func (s *GenericStore[T, PT]) GetIDs(ctx context.Context) ([]string, error) {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.GetAll)
+	var sacQueryFilter *v1.Query
+	if s.hasPermissionsChecker() {
+		if ok, err := s.permissionChecker.GetAllowed(ctx); err != nil || !ok {
```

Comment:
In the current state of existing `PermissionChecker` implementation, it does not make any difference, as these will check for global read access to the underlying resource(s).
Nevertheless, the generated store code was calling `GetIDsAllowed` on the permission checker.